lib/matplotlib/contour.py

```diff
@@ -977,6 +977,10 @@ def collections(self):
                 self.axes.add_collection(col)
         return self._old_style_split_collections
 
+    def set_paths(self, paths):
+        self._paths = paths
+        self.stale = True
```

Comment:
I agree with @anntzer's comment https://github.com/matplotlib/matplotlib/issues/26340#issuecomment-1639611508
I think we could just add this change to `Collection.set_paths(paths)` method definition instead so others can also subclass and take advantage of it (could remove the method definition from `PathCollection` then too I think.

Reply:
Ah sorry, I misread @anntzer's comment.  I've now updated `Collection.set_paths`.  I left my test in because `set_paths` does not appear in `test_collection.py`.  I assume the `ContourSet` is as good a place as any to make sure this code path is covered?